python/taichi/lang/matrix.py

```diff
@@ -1045,13 +1045,488 @@ def identity(dt, n):
+            >>> ti.Matrix.translate(1, 2, 3)
+            [[ 1 0 0 1]
+             [ 0 1 0 2]
+             [ 0 0 1 3]
+             [ 0 0 0 1]]
+        """
+        return Matrix([[1., 0., 0., dx], [0., 1., 0., dy], [0., 0., 1., dz],
+                       [0., 0., 0., 1.]])
+
+    @staticmethod
+    def scale(sx, sy, sz):
+        """Constructs a scale Matrix with shape (4, 4).
+
+        Args:
+            sx (float): scale x.
+            sy (float): scale y.
+            sz (float): scale z.
+
+        Returns:
+            :class:`~taichi.Matrix`: An `4 x 4` translate matrix.
+
+        Example:
+
+            >>> import math
+            >>> ti.Matrix.scale(1, 2, 3)
+            [[ 1 0 0 0]
+             [ 0 2 0 0]
+             [ 0 0 3 0]
+             [ 0 0 0 1]]
+        """
+        return Matrix([[sx, 0., 0., 0.], [0., sy, 0., 0.], [0., 0., sz, 0.],
+                       [0., 0., 0., 1.]])
+
+    @staticmethod
+    def eulerAngleX(angleX):
+        """Creates a 3D 4 * 4 homogeneous rotation matrix from an euler angle X.
+
+        Args:
+            angleX (float): angle in radians unit around X axis
+
+        """
+        cosX = ops_mod.cos(angleX)
+        sinX = ops_mod.sin(angleX)
+
+        return Matrix([[1., 0., 0., 0.], [0., cosX, sinX, 0.],
+                       [0., -sinX, cosX, 0.], [0., 0., 0., 1.]])
+
+    @staticmethod
+    def eulerAngleY(angleY):
+        """Creates a 3D 4 * 4 homogeneous rotation matrix from an euler angle Y.
+
+        Args:
+            angleY (float): angle in radians unit around Y axis
+
+        """
+        cosY = ops_mod.cos(angleY)
+        sinY = ops_mod.sin(angleY)
+
+        return Matrix([[cosY, 0., -sinY, 0.], [0., 1., 0., 0.],
+                       [sinY, 0., cosY, 0.], [0., 0., 0., 1.]])
+
+    @staticmethod
+    def eulerAngleZ(angleZ):
+        """Creates a 3D 4 * 4 homogeneous rotation matrix from an euler angle Z.
+
+        Args:
+            angleZ (float): angle in radians unit around Z axis
+
+        """
+        cosZ = ops_mod.cos(angleZ)
+        sinZ = ops_mod.sin(angleZ)
+
+        return Matrix([[cosZ, sinZ, 0., 0.], [-sinZ, cosZ, 0., 0.],
+                       [0., 0., 1., 0.], [0., 0., 0., 1.]])
+
+    @staticmethod
+    def eulerAngleXY(angleX, angleY):
+        """Creates a 3D 4 * 4 homogeneous rotation matrix from euler angles(X * Y).
+
+        Args:
+            angleX (float): angle in radians unit around X axis
+            angleY (float): angle in radians unit around Y axis
+
+        """
+        cosX = ops_mod.cos(angleX)
+        sinX = ops_mod.sin(angleX)
+        cosY = ops_mod.cos(angleY)
+        sinY = ops_mod.sin(angleY)
+
+        return Matrix([[cosY, -sinX * -sinY, cosX * -sinY, 0.],
+                       [0., cosX, sinX, 0.],
+                       [sinY, -sinX * cosY, cosX * cosY, 0.], [0., 0., 0.,
+                                                               1.]])
+
+    @staticmethod
+    def eulerAngleYX(angleY, angleX):
```

Comment:
Can we make the order an argument like `rotation3d_euler_angle(xrad=30.0, yrad=30.0, zrad=30.0, order="xyz")` so that there won't be a bunch of similar APIs bloated in the doc site?

Reply:
Not agree with this point of view. 
1. `eulerAngleYX` etc. support both taichi scope and python scope, string value is not a good thing for taich sope
2. There are too many arguement for python users, and passing string is also strange.
3. I think the name of API like `eulerAngleYX` is clearly enough for users (first rotate by X axis, then rotate by Y axis), and the AngleX and AngleY are also clearly.
4. the name of the rotation APIs is the same with glm, and is good for those who are familar with glm

Reply:
I see, the string arg was a bad idea. but I still expect a simpler set of APIs.

1. Named after `rotation3d` to be consistent with `rotation2d`;
2. 1-/2-axis rotation APIs can be removed because you can give zero values to the unused parameters of the 3-axis APIs;
3. `rotate_by_vector` , i.e. axis-angle rotation description, should be the default rotation mode, and I expect a name of `rotation3d` with no suffix;
4. Signify param names that the inputs are in *radian*;
5. Better to provide an APIT to ingest quaternions, but maybe not in this PR.